### DISCUSSION TITLE
The usage of the Slash Command extension

### DISCUSSION THREAD

**Discussion by boundless-forest** [zed-industries/zed#17403] [Feature Requests]
I noticed that Zed now supports three types of extensions: language, theme, and slash. After reading https://zed.dev/docs/extensions/slash-commands, I'm curious about what kind of extensions could be built based on the slash command. It seems that it cannot interact with the editor at this time; it only accepts user input in the assistant panel. It resembles a built-in CLI tool in Zed.

Are there any workable extensions available that are built based on this? I am eager to learn more about it.

**Comment by dream-dasher** [comment]
[Example Slash Command extension](https://github.com/zed-industries/zed/tree/main/extensions/slash-commands-example)


Their docs could def use work (I don't recall it even being clear what slash commands were when I read it -- had to do some searching -- I thought it was some task runner that wasn't popping up because `/` was used by vim) -- but the idea is that they are convenience functions specifically for interacting with the LLM assistant.  
e.g. there's a built-in slash command to pull in terminal output or to pull in current open tab.

One could imagine doing other, space-optimized, slash commands:
For example, pulling in all the git commit messages from a certain time frame.  Or pulling in two different branches of a document's current state.

If one has stylistic or method preferences, or things to avoid, one could statically pull from some maintained file and populate that.

Seems like it's basically convenience functions for building context.  (But I haven't delved much -- I've only just begun looking at the assistant stuff.)

[You may know all this now, but I came across the Q while trying to figure out what slash commands were and figured I’d answer]

**Comment by artivilla** [comment]
@boundless-forest if you are interested in building something, cursor  has some interesting ones such as @docs where you can index docs from third party sites. I can do @docs https://ui.shadcn.com/ and it will index all pages in the sitemaps vs /fetch which only does a single url.

throwing Ideas as I'd love this feature.
